### PR TITLE
add new metadata docs

### PR DESCRIPTION
**Pull request by EugenWay**
Updated sections:

-
-
-

### CODE REVIEW

File: docs/developing-contracts/metadata.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+```
+
+## Verify metadata
+
+To make it possible to verify metadata for a program, you can use `metahash()` function. It allows verify metadata on-chain.
```
Comment (shamilsan) [inline docs/developing-contracts/metadata.md:65]:
```suggestion
To make it possible to verify metadata for a program, you can use the `metahash()` function. It allows verifying metadata on-chain.
```

File: docs/developing-contracts/metadata.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+```rust
+#[no_mangle]
+// It returns the Hash of metadata. 
+// .metahash is generating automatically while you using build.rs
```
Comment (shamilsan) [inline docs/developing-contracts/metadata.md:70]:
```suggestion
// .metahash is generating automatically while you are using build.rs
```

File: docs/developing-contracts/state.md [hunk 3]
```diff
@@ -8,52 +8,96 @@ sidebar_position: 3
...
+#[derive(TypeInfo, Decode, Encode, Clone)]
+pub struct Wallet {
+    pub id: ActorId,
+    pub person: String,
 }
 
-// initialize state itself
-static mut STATE: State = State::new();
+
+// declare and initialize State
```
Comment (shamilsan) [inline docs/developing-contracts/state.md:20]:
```suggestion
// declare and initialize the state
```

File: docs/developing-contracts/state.md [hunk 4]
```diff
@@ -8,52 +8,96 @@ sidebar_position: 3
...
-}
 
-impl State {
-    // Create a state
-    pub const fn new() -> Self {
-        Self {
-            votes_received: BTreeMap::new(),
-        }
-    }
+// describe State structure
```
Comment (shamilsan) [inline docs/developing-contracts/state.md:12]:
```suggestion
// describe state structure
```

File: docs/developing-contracts/state.md [hunk 5]
```diff
@@ -8,52 +8,96 @@ sidebar_position: 3
...
-#[derive(Encode, Decode, TypeInfo)]
-pub struct ContractState {}
+#[no_mangle]
+extern "C" fn state() {
+    msg::reply(unsafe { WALLETS.clone() }, 0).expect("Failed to share state");
+}
+```
 
-static mut STATE: ContractState = ContractState {};
+By default, the `state()` function returns the full State of the contact.
```
Comment (shamilsan) [inline docs/developing-contracts/state.md:43]:
```suggestion
By default, the `state()` function returns the whole state of the contact.
```

File: docs/developing-contracts/state.md [hunk 6]
```diff
@@ -8,52 +8,96 @@ sidebar_position: 3
...
-    let query =
-        String::from_utf8(msg::load_bytes().expect("Unable to load bytes")).expect("Invalid query");
-    let reply = if query == "CONTRACT_STATE" {
-        let encoded = unsafe { STATE.encode() };
-        gstd::util::to_leak_ptr(encoded)
-    } else {
-        gstd::util::to_leak_ptr(vec![])
-    };
-    reply
+## Custom program to read State
```
Comment (shamilsan) [inline docs/developing-contracts/state.md:45]:
```suggestion
## Custom program to read the state
```

File: docs/developing-contracts/state.md [hunk 7]
```diff
@@ -8,52 +8,96 @@ sidebar_position: 3
...
+
+Additionally, you can create your own program for read State. This wrapper will allow you to implement your custom functions for the client side, not depending on the main program.
```
Comment (shamilsan) [inline docs/developing-contracts/state.md:47]:
```suggestion
Additionally, you can create your own program to read the state. This wrapper will allow you to implement custom functions for the client side, not depending on the main program.
```

File: docs/developing-contracts/state.md [hunk 8]
```diff
@@ -8,52 +8,96 @@ sidebar_position: 3
...
+
+This has a number of advantages, for example, you will always be able to read the State even if the program changes (provided that the incoming or outgoing types have not changed).
```
Comment (shamilsan) [inline docs/developing-contracts/state.md:49]:
```suggestion
This has a number of advantages, for example, you will always be able to read the state even if the program changes (as long as the incoming or outgoing types have not changed).
```

File: docs/developing-contracts/state.md [hunk 9]
```diff
@@ -8,52 +8,96 @@ sidebar_position: 3
...
+
+To do this, we need to create an independent program. And to describe the necessary functions inside `metawasm` trait. For example:
```
Comment (shamilsan) [inline docs/developing-contracts/state.md:51]:
```suggestion
To do this, we need to create an independent program and describe the necessary functions inside the `metawasm` trait. For example:
```

**Review by shamilsan — APPROVED** [review]
Awesome 🔥 

My comments are for consideration only; not all of them should be accepted.

**Review by AndrePanin — APPROVED** [review]

**Pull request merged**